### PR TITLE
Remove Component namespace on state

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

**Comment by loic425** [comment]
@diimpp Also introduced in 1.11 alpha

File: src/Component/tests/spec/State/ProviderSpec.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ function it_throws_an_exception_when_configured_provider_is_not_a_provider_insta
         $locator->has('\stdClass')->willReturn(true);
         $locator->get('\stdClass')->willReturn(new \stdClass());
 
-        $this->shouldThrow(new \InvalidArgumentException('Expected an instance of Sylius\Component\Resource\State\ProviderInterface. Got: stdClass'))
+        $this->shouldThrow(new \InvalidArgumentException('Expected an instance of Sylius\Resource\State\ProviderInterface. Got: stdClass'))
```
Comment (diimpp) [inline src/Component/tests/spec/State/ProviderSpec.php:76]: As idea
```suggestion
        $this->shouldThrow(new \InvalidArgumentException(sprintf('Expected an instance of %s. Got: %s', ProviderInterface::class, \stdClass::class)))
```

File: src/Component/tests/spec/State/ProcessorSpec.php [hunk 2]
```diff
@@ -73,7 +73,7 @@ function it_throws_an_exception_when_configured_processor_is_not_a_processor_ins
         $locator->has('\stdClass')->willReturn(true);
         $locator->get('\stdClass')->willReturn(new \stdClass());
 
-        $this->shouldThrow(new \InvalidArgumentException('Expected an instance of Sylius\Component\Resource\State\ProcessorInterface. Got: stdClass'))
+        $this->shouldThrow(new \InvalidArgumentException('Expected an instance of Sylius\Resource\State\ProcessorInterface. Got: stdClass'))
```
Comment (diimpp) [inline src/Component/tests/spec/State/ProcessorSpec.php:76]: As idea
```suggestion
        $this->shouldThrow(new \InvalidArgumentException(sprintf('Expected an instance of %s. Got: %s', ProcessorInterface::class, \stdClass::class)))
```

File: src/Component/tests/spec/State/ResponderSpec.php [hunk 3]
```diff
@@ -11,15 +11,15 @@
 
 declare(strict_types=1);
 
-namespace spec\Sylius\Component\Resource\State;
+namespace spec\Sylius\Resource\State;
```
Comment (diimpp) [inline src/Component/tests/spec/State/ResponderSpec.php:14]: Missing case for `Assert`, like for processor and provider

```patch
+++ function it_throws_an_exception_when_configured_responder_is_not_a_responder_instance()
```

**Review by diimpp — APPROVED** [review]

File: src/Component/tests/spec/State/ProviderSpec.php (hunk 1 above)
Comment (loic425) [reply src/Component/tests/spec/State/ProviderSpec.php:76]: Yeah, I think I've read that it's better not to use constants and things like that in specs/tests.
@lchrusciel @Zales0123?

File: src/Component/tests/spec/State/ResponderSpec.php (hunk 3 above)
Comment (loic425) [reply src/Component/tests/spec/State/ResponderSpec.php:14]: fixed

**Review by diimpp — APPROVED** [review]

File: src/Component/tests/spec/State/ProviderSpec.php (hunk 1 above)
Comment (lchrusciel) [reply src/Component/tests/spec/State/ProviderSpec.php:76]: I don't have a strong opinion about that. Will the test still make sense if we would move provider to other namespace? Yup. Both approaches are fine for me

**Comment by lchrusciel** [comment]
Thank you, @loic425!

**Pull request merged**